File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.rs
impl Solution {
    pub fn max_profit(prices: Vec<i32>) -> i32 {
        let (mut max_profit, mut lowest_price) = (0, 100_000);
        
        for (day, price) in prices.iter().enumerate() {
            if price < &lowest_price {
                lowest_price = *price;
            }
            
            let profit = *price - lowest_price;

            max_profit = max_profit.max(profit);
            
        }
        
        max_profit
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.ts
function maxProfit(prices: number[]): number {
    let lowest = 1_000_000; let maxProfit = 0;

    for (let i = 0; i < prices.length; i++) {
        let price = prices[i];
        if (price < lowest) {
            lowest = price;
        }
        let profit = price - lowest;
        maxProfit = Math.max(profit, maxProfit);
    }
    return maxProfit;
};